src/index.ts

```diff
@@ -1,16 +1,50 @@
+import { Fraction, transitiveOrderbook } from '@gnosis.pm/dex-contracts'
 import express from 'express'
+import Web3 from 'web3'
+import { CategoryServiceFactory, CategoryConfiguration, Category, LogLevel } from 'typescript-logging'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import * as yargs from 'yargs'
+
+const argv = yargs
+  .option('node', {
+    describe: 'RPC endpoint to connect to',
+    demand: true,
+  })
+  .option('pageSize', {
+    describe: 'The number of orders to fetch per page',
+    default: 50,
+  })
+  .option('port', {
+    describe: 'Port to bind on',
+    default: '8080',
+  })
+  .option('verbosity', {
+    describe: 'log level',
+    choices: ['TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'],
+    default: 'INFO',
+  }).argv
+
 const app = express()
-const port = 8080 // default port to listen
+const web3 = new Web3(argv.node as string)
+
+CategoryServiceFactory.setDefaultConfiguration(new CategoryConfiguration(LogLevel.fromString(argv.verbosity)))
+const logger = CategoryServiceFactory.getLogger(new Category('dex-price-estimation'))
+const orderbooksFetcher = new OrderbookFetcher(web3, argv.pageSize, logger)
 
 /* tslint:disable:no-unused-expression */
 
-// define a route handler for the default home page
-app.get('/', (_, res) => {
-  res.send('Hello world!')
+app.get('/orderbook', (req, res) => {
+  const transitive = transitiveOrderbook(orderbooksFetcher.orderbooks, req.query.base, req.query.quote, 2)
+  res.json(JSON.stringify(transitive))
+})
+
+app.get('/price', (req, res) => {
+  const transitive = transitiveOrderbook(orderbooksFetcher.orderbooks, req.query.base, req.query.quote, 2)
+  const exactPrice = transitive.priceToSellBaseToken(req.query.sell)
+  const priceWithRoundingBuffer = exactPrice?.mul(new Fraction(999, 1000))
```

Comment:
magic numbers.
Also, a comment is desirable for why you reduce the price a percentage.
Are you taking fee into consideration?

Reply:
Yes, however even with fees the solver adds some extra rounding in order to be able to convert from its internal float representation back into u256 while making the equality constraints in the smart contract exact (sellTokens == feeTokens to the WEI).

Thus we need to add another small rounding margin to ensure that the solver will be able to find a solution. Will also put this into a configuration param